pkg/oc/cli/cmd/rollout/cancel.go

```diff
@@ -55,40 +64,49 @@ event will be emitted.`)
   %[1]s rollout cancel dc/nginx`)
 )
 
-func NewCmdRolloutCancel(fullName string, f kcmdutil.Factory, out io.Writer) *cobra.Command {
-	opts := &CancelOptions{}
+func NewRolloutCancelOptions(streams genericclioptions.IOStreams) *CancelOptions {
+	return &CancelOptions{
+		IOStreams:  streams,
+		PrintFlags: genericclioptions.NewPrintFlags("already cancelled"),
+	}
+}
+
+func NewCmdRolloutCancel(fullName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewRolloutCancelOptions(streams)
+
 	cmd := &cobra.Command{
 		Use:     "cancel (TYPE NAME | TYPE/NAME) [flags]",
 		Long:    rolloutCancelLong,
 		Example: fmt.Sprintf(rolloutCancelExample, fullName),
 		Short:   "cancel the in-progress deployment",
 		Run: func(cmd *cobra.Command, args []string) {
-			kcmdutil.CheckErr(opts.Complete(f, cmd, out, args))
-			kcmdutil.CheckErr(opts.Run())
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
 		},
 		ValidArgs: []string{"deploymentconfig"},
 	}
+
+	o.PrintFlags.AddFlags(cmd)
+
 	usage := "Filename, directory, or URL to a file identifying the resource to get from a server."
-	kcmdutil.AddFilenameOptionFlags(cmd, &opts.FilenameOptions, usage)
+	kcmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, usage)
 	return cmd
 }
 
-func (o *CancelOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, out io.Writer, args []string) error {
-	var err error
-
+func (o *CancelOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
 	if len(args) == 0 && len(o.FilenameOptions.Filenames) == 0 {
 		return kcmdutil.UsageErrorf(cmd, cmd.Use)
```

Comment:
Make this error more obvious what's missing by explicitly telling what's required.